nextest-runner/src/list/test_list.rs

```diff
@@ -882,6 +894,50 @@ pub(crate) fn make_test_command(
 
     let mut cmd = std::process::Command::new(program);
 
+    let make_windows_compatible_key = |key: OsString| {
+        // On Windows, environment variable keys are case-insensitive, so create
+        // a canonical form to check existence.
+        if cfg!(windows) {
+            key.to_ascii_lowercase()
+        } else {
+            key
+        }
+    };
+
+    // NB: we will always override user-provided environment variables with the
+    // ones we set below.
+    let mut existing_keys: HashSet<_> = std::env::vars_os()
+        .map(|(k, _v)| k)
+        .map(make_windows_compatible_key)
+        .collect();
+    for CargoEnvironmentVariable {
+        source,
+        name,
+        value,
+        force,
+        relative,
+    } in env
+    {
+        let name_os_string = make_windows_compatible_key(OsString::from(name));
+        if !existing_keys.insert(name_os_string) && !force {
+            continue;
+        }
+
+        let value = if *relative {
+            let base_path = match source {
+                Some(source_path) => source_path,
+                None => panic!("Cannot use a relative path for environment variable {name:?} whose source is not a config file"),
+            };
+            match base_path.parent() {
+                Some(base_path) => base_path.join(value).into_string(),
+                None => value.clone(),
```

Comment:
This needs one more level of `parent()`: https://doc.rust-lang.org/cargo/reference/config.html#env

> Setting the relative flag evaluates the value as a config-relative path *that is relative to the parent directory of the .cargo directory* that contains the config.toml file

Going to fix this.